ignite/distributed/utils.py

```diff
@@ -637,3 +639,31 @@ def wrapper(*args: Any, **kwargs: Any) -> Optional[Any]:
         return wrapper
 
     return _one_rank_only
+
+
+@contextmanager
+def one_process_first(rank: int = 0) -> Any:
+    """This is a context manager function that ensures a specific process runs first before others in a distributed
+    environment.
+
+    Args:
+        rank: an integer that specifies the rank of the process that should execute the code
+    block inside the context manager first.
+
+    Examples:
+        .. code-block:: python
+
+            def download_dataset():
+                ...
+
+            with one_process_first(rank=0):
```

Comment:
```suggestion
            with one_process_first():
```
Maybe, we can omit default value